azure/services/managedclusters/spec.go

```diff
@@ -581,3 +591,20 @@ func computeDiffOfNormalizedClusters(managedCluster containerservice.ManagedClus
 	diff := cmp.Diff(clusterNormalized, existingMCClusterNormalized)
 	return diff
 }
+
+func getIdentityProfile(identityProfile *infrav1.IdentityProfile) (managedClusterIdentity *containerservice.ManagedClusterIdentity, err error) {
+	if identityProfile.Type != "" {
+		managedClusterIdentity = &containerservice.ManagedClusterIdentity{
+			Type: containerservice.ResourceIdentityType(identityProfile.Type),
+		}
+		if managedClusterIdentity.Type == containerservice.ResourceIdentityTypeUserAssigned {
+			if identityProfile.UserAssignedIdentityResourceID == "" {
+				err = errors.Errorf("IdentityProfile is set to \"UserAssigned\" but no UserAssignedIdentityResourceID is present")
+				return
+			}
+			managedClusterIdentity.UserAssignedIdentities = make(map[string]*containerservice.ManagedClusterIdentityUserAssignedIdentitiesValue, 1)
+			managedClusterIdentity.UserAssignedIdentities[identityProfile.UserAssignedIdentityResourceID] = &containerservice.ManagedClusterIdentityUserAssignedIdentitiesValue{}
```

Comment:
nit: inlining this might be slightly less verbose? At the expense of some possible memory optimization Go does with `make(..., 1)` that doesn't happen with a literal.
```suggestion
			managedClusterIdentity.UserAssignedIdentities = map[string]*containerservice.ManagedClusterIdentityUserAssignedIdentitiesValue{
				identityProfile.UserAssignedIdentityResourceID: {},
			}
```

Reply:
deal